buckets/buckets.go

```diff
@@ -270,9 +320,14 @@ func (b *Buckets) Delete(ctx context.Context, dbID thread.ID, key string, opts .
 }
 
 // Archive pushes the current root Cid to the corresponding FFS instance of the bucket.
-// If it's not the first archive, we leverage the Replace() API to more efficient pinning.
-// The operation is async, so it will return as soon as the archiving is scheduled in Powergate.
-func (b *Buckets) Archive(ctx context.Context, dbID thread.ID, key string, c cid.Cid, opts ...Option) error {
+// The behaviour changes depending on different cases, depending on a previous archive.
+// 0. No previous archive or last one aborted: simply pushes the Cid to the FFS instance.
+// 1. Last archive exists with the same Cid:
+//   a. Last archive Successful: fails, there's nothing to do.
+//   b. Last archive Executing/Queued: fails, that work already starting and is in progress.
+//   c. Last archive Failed/Canceled: work to do, push again with override flag to try again.
+// 2. Archiving on new Cid: work to do, it will always call Replace(,) in the FFS instance.
```

Comment:
Changes in `Archive` are mostly to have well-defined behavior regarding multiple scenarios.
The comment tries to be quite explicit about the scenarios and what the logic decides to do. I think these are reasonable behaviors for each case.

I'm planning to switch each error case in a sentinel error, and permeate them in a reasonable way through the API, so the CLI can show a nicer message in each case. Left that for the next PR since it involves looking at a reasonable way to check controlled errors in gRPC in this API. (We somewhat chatted about that before)

Reply:
Sweet